### DISCUSSION TITLE
Time Format Vote

### DISCUSSION THREAD

**Discussion by Jyers** [Jyers/Marlin#117] [Q&A] [answered]
### 00:00 Format: 🚀

**Comment by Henri-J-Norden** [comment]
The claim of 00h00m being "more universal and human readable" is not based on fact. 00:00 is the international standard representation for time as specified by [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) and is perfectly human readable.

**Reply by tiagofreire-pt** [reply]
> 
> 
> The claim of 00h00m being "more universal and human readable" is not based on fact. 00:00 is the international standard representation for time as specified by [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) and is perfectly human readable.

The proposed format for duration (HH h mm m) is also a **valid** standard and defined in the **ISO 8601 itself**: https://en.wikipedia.org/wiki/ISO_8601#Durations

**Reply by Henri-J-Norden** [reply]
Oh interesting, I actually did not know that, as I don't think I've ever seen it used.

**Comment by tiagofreire-pt** [comment]
Hi,

Yes, the **time** representation is standardized as HH : mm, with ISO 8601.

What I'm proposing here is not a representation of time itself, as an instant moment of time, but **duration** or **accumulated amount/elapsed/remaining.**

The proposed format for **duration** (HH **h** mm **m**) is also a valid standard and defined in the ISO 8601 itself: https://en.wikipedia.org/wiki/ISO_8601#Durations

Even further, is the format chosen by the 2 most predominant slicers to represent printing duration:

Cura:

![cura](https://user-images.githubusercontent.com/41837236/108980472-9c4cbc00-7683-11eb-82dd-2ccdb6441d37.png)

Prusa Slicer:

![prusaslicer](https://user-images.githubusercontent.com/41837236/108979983-1fb9dd80-7683-11eb-9fb0-3666d2e1a334.JPG)

Proposal:

![ender3v2](https://user-images.githubusercontent.com/41837236/109000831-1ab55800-769c-11eb-818a-18f39ffcfc2f.png)

I hope that we could all agree on this. :)

Thanks.

**Comment by Cal-LF** [comment]
I can function with either style.

**Comment by amandamana** [comment]
00h00m appears less readable to me because the 'h' and 'm' characters are much wider on the screen than a colon, which makes it more difficult to visually chunk the numbers. The low resolution and narrow viewing angle of the V2 screen don't help. I also think that shorter strings are easier to read, and that either 'h' or 'm' is redundant in view of the other.

Also note that for the proposed duration to comply with ISO 8601, it needs period and time designators, such as PT00H00M.

I don't think the issue is human readability, but possible ambiguity about whether 00:00 is hh:mm or mm:ss. Of course, that's easy to figure out when looking at a live display.

When possible, I'd like to see h:mm:ss (no leading zero for h) as the most informative, unambiguous format.

**Comment by Legsmaniac** [comment]
@tiagofreire-pt  Here's another one for you.

Have just finished a nearly 8 hour print.
Using the latest Jyers flavour, I'd forgotten I'd set my config to tell me total finished print time.
Here's what I see..........

![PrintTime](https://user-images.githubusercontent.com/50076851/109046910-2bca8d00-76cd-11eb-821a-a1694439820b.jpg)

So there's another example of time format, the format which is being proposed.
I say yes, I'd happily go with it.    😉

**Reply by amandamana** [reply]
I think that's an example of how space characters are added because 7h43m37s runs together with poor readability on the display. That solution helps, but it's not always possible when screen space is at a premium. And even with the spaces, I think 7: would be easier to read off angle than 7h.

**Comment by Henri-J-Norden** [comment]
It would be possible to support both formats, by simply adding a "Time format" toggle in one of the menus. I have been working on a leveling menu for UBL with a mesh viewer, so extending the menu is pretty straightforward. I have not, however, looked into how to add a new EEPROM variable, which would be necessary in this case.

**Reply by Jyers** [reply]
This was actually what I was planning on doing considering the situation. I was also a little turned off by the eeprom aspect, but if  have time soon I'm sure it would not be the worst thing to figure out.

**Comment by Jyers** [comment, marked as answer]
At some point I will hopefully implement both of these formats and provide a setting in advanced options to choose. That is when I figure out how to read and write to eeprom...
